cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -128,5 +130,45 @@
    * @param numPartitions the minimum number of new partitions added to the dataset to trigger the schedule
    * @return this {@link ScheduleBuilder}
    */
-  ScheduleCreationSpec triggerOnPartitions(String datasetName, int numPartitions);
+  ScheduleCreationBuilder triggerOnPartitions(String datasetName, int numPartitions);
```

Comment:
also add a method that accepts namespaceId. Update javadoc for this method to reflect that the dataset is in the same namespace as the app.